docs/troubleshooting.md

```diff
@@ -0,0 +1,202 @@
+
+For more troubleshooting tips, see the official k8s docs for [troubleshooting pods](https://kubernetes.io/docs/tasks/debug/debug-application/debug-pods/).
+
+### Troubleshoot Services
+
+Sometimes, when you start a new installation in Kubernetes you may find that a service doesn’t respond when you try to access it, although it was created by a pod in a deployment.
+
+First you can check that the service you are trying to access actually exists, by running the following (replace SVC-NAME with the name of the service you want to access):
+
+```shell
+$ kubectl get SVC-NAME
+```
+
+To get detailed information around the service if it exists, run:
+
+```shell
+$ kubectl describe SVC-NAME
+```
+
+You should see an output message providing some information including any service issues.
+
+If the service is registered and no issues were detected from the step above, it could also be a DNS problem. DNS problems
+typically occurs when a service is in a different namespace then the pod or if the cluster does not have DNS enabled.
+
+To troubleshoot, run the *kubectl get pods* command to get the name of your pod. Then run the command below, this will give you an indication if the DNS resolution is working or not. (remember to replace the `POD-NAME` placeholder with the pod name and `SVC-NAME` placeholder with the name of the service).
+
+```shell
+$ kubectl exec -it POD-NAME nslookup SVC-NAME
+```
+
+If the error persists, then [confirm that DNS is enabled for your Kubernetes cluster](https://kubernetes.io/docs/tasks/administer-cluster/dns-custom-nameservers/). Check out the Kubernetes official documentation to learn how to [debug DNS resolution](https://kubernetes.io/docs/tasks/administer-cluster/dns-debugging-resolution/).
+
+Lastly, if the service was attached to an [ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) (Load Balancer), you can check for issues related to the ingress deployment. To troubleshoot, run the *kubectl get ingress* command to get the name of your ingress. Then, replacing `INGRESS-NAME` placeholder with the ingress name, run:
+
+```shell
+$ kubectl describe ingress INGRESS-NAME
+```
+
+You should see an output message providing some information including any ingress issues.
+
+For more troubleshooting tips, see the official k8s docs for [troubleshooting services](https://kubernetes.io/docs/tasks/debug/debug-application/debug-service/).
+
+### Troubleshoot Persistence Volumes
+
+The next common problem to occur is Persistent Volumes failing to provision. To check the status of your persistent volumes, run *kubectl get pvc*. If the output message shows that your PVC status is pending, this may be because your cluster does not support dynamic provisioning (such as a bare metal cluster). In this case, the pod is unable to start because the cluster is unable to fulfil the request to attach the persistent volume to your pod.
```

Comment:
you want to add some info around PVs as well? You focus on the PVCs here and it might be that the claim can not be done successfully because the PV has issues...?

Reply:
Updated this whole section to include some background on the 3 main types StorageClass, Persistent Volume, and Persistent Volume Claim and going through and checking each phase